File: jupyter2/problems/iteration/gaussian_integral/ml/gaussian_integral.ml
(** begin my answer *)

(** end my answer *)

let main () =
  assert (abs_float (gaussian_integral  1.0 1000)  -. 1.493648 < 1.0e-6);
  assert (abs_float (gaussian_integral  2.0 2000)  -. 1.764163 < 1.0e-6);
  assert (abs_float (gaussian_integral 10.0 10000) -. Float.sqrt Float.pi < 1.0e-6);
  Printf.printf "OK\n"
;;

main()

File: jupyter2/ps/iteration/gaussian_integral/go/gaussian_integral.go
package main
import "math"
import "fmt"

/** begin my answer */

/** begin hidden */
func gaussianIntegral(a float64, n int64) float64 {
	dx := 2.0 * a / float64(n)
	sum := 0.0
	for i := int64(0); i < n; i++ {
		x := -a + float64(i)*dx
		sum += math.Exp(-x*x) * dx
	}
	return sum
}
/** end hidden */
/** end my answer */

func main() {
	if !(math.Abs(gaussianIntegral(1, 1000) - 1.493648) < 1.0e-6) { panic("wrong") }
	if !(math.Abs(gaussianIntegral(2, 2000) - 1.764163) < 1.0e-6) { panic("wrong") }
	if !(math.Abs(gaussianIntegral(10, 10000) - math.Sqrt(math.Pi)) < 1.0e-6) { panic("wrong") }
	fmt.Println("OK")
}
